server/schedule/checker/rule_checker.go

```diff
@@ -30,19 +33,23 @@ import (
 
 // RuleChecker fix/improve region by placement rules.
 type RuleChecker struct {
-	cluster           opt.Cluster
-	ruleManager       *placement.RuleManager
-	name              string
-	regionWaitingList cache.Cache
+	cluster              opt.Cluster
+	ruleManager          *placement.RuleManager
+	name                 string
+	regionWaitingList    cache.Cache
+	offlineLeaderCounter map[uint64]uint64
+	lastReplaceOpTime    time.Time
 }
 
 // NewRuleChecker creates a checker instance.
 func NewRuleChecker(cluster opt.Cluster, ruleManager *placement.RuleManager, regionWaitingList cache.Cache) *RuleChecker {
 	return &RuleChecker{
-		cluster:           cluster,
-		ruleManager:       ruleManager,
-		name:              "rule-checker",
-		regionWaitingList: regionWaitingList,
+		cluster:              cluster,
+		ruleManager:          ruleManager,
+		name:                 "rule-checker",
+		regionWaitingList:    regionWaitingList,
+		offlineLeaderCounter: make(map[uint64]uint64),
+		lastReplaceOpTime:    time.Now(),
```

Comment:
How about changing to `lastUpdateTime` and put it with offlineLeaderCounter together?